hookd/templates/form.html

```diff
@@ -1,37 +1,168 @@
 {{define "body"}}
-<p>
-    Authenticated as <span class="name">{{.User.Name}}</span> (<span class="login">@{{.User.Login}}</span>), <a
-        href="/auth/logout">sign out</a>.
-</p>
-<h2>Repositories</h2>
-{{if .Repository}}
-<form method="POST" action="/auth/submit">
-    <input readonly name="repository" value="{{.Repository}}">
-
-    {{if .Teams}}
-    <h2>Teams</h2>
-    <select multiple name="teams">
-        {{range .Teams}}
-        <option value="{{.Name}}">{{.Name}}</option>
-        {{end}}
-    </select>
-    <br>
-    <button type="submit">Grant access</button>
-    {{else}}
-    <p>No teams exist where you are listed as an <strong>admin</strong>.</p>
-    {{end}}
-</form>
-
-{{else if .Repositories}}
-<form>
-    <select name="repository" onChange="this.form.submit()">
-        <option>Select repository</option>
-        {{range .Repositories}}
-        <option value="{{.Name}}">{{.FullName}}</option>
-        {{end}}
-    </select>
-</form>
-{{else}}
-<p>No repositories exist where you are listed as an <strong>admin</strong>.</p>
-{{end}}
+<div id="page-container">
+    <p id="authenticated">
+        Authenticated as <span class="name">{{.User.Name}}</span> (<span class="login">@{{.User.Login}}</span>), <a
+            href="/auth/logout">sign out</a>.
+    </p>
+
+    <div id="repository-spinner">
+        <p>Fetching repositories from GitHub</p>
+        <div class="spinner"></div>
+    </div>
+
+    <div id="form-container"></div>
+</div>
+
+<script>
+    function getJson(path, success, error) {
+        var xhr = new XMLHttpRequest();
+        xhr.onload = function () {
+            if (xhr.status === 200) {
+                success(JSON.parse(xhr.responseText));
+            } else {
+                error(xhr);
+            }
+        };
+        xhr.open("GET", path);
+        xhr.send();
+    }
+
+    function fetchRepositoryTeams() {
+        removeElement(
+            document.getElementById("team-list-container"),
+            document.getElementById("team-error")
+        );
+
+        var select = document.getElementById("repository-select")
+        var repository = select.options[select.selectedIndex].value;
+
+        if (!repository) {
+            return;
+        }
+
+        var teamsSpinnerText = document.createElement("p");
```

Comment:
All good so far. But here it starts to get a bit crazy when you programatically create DOM elements. Isn't there another way?

Reply:
Well, we could have them in the DOM and move them around with JS instead and swap visibility along the way. Or we could use a lib that makes creating elements a lot easier than with vanilla JS.

Reply:
I can add a commit that instead of creating elements using JS just moves existing elements around.